ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -324,6 +336,36 @@ void fillChunks() {
   }
 
   /**
+   * Called whenever the channel has data but no free or building chunk is available to be filled.
+   */
+  private void maybeStartTrackingWaitTime() {
+    if (startTimeForChunkAvailabilityWaitMs == 0) {
+      // this is the first point in time after the last chunk filling (if any) when the filling was blocked due to
+      // chunk unavailability, so mark this time.
+      startTimeForChunkAvailabilityWaitMs = time.milliseconds();
+    }
+  }
+
+  /**
+   * Called whenever the channel has data and there is a free or building chunk available to be filled.
+   */
+  private void maybeUpdateTrackedWaitTime() {
+    if (startTimeForChunkAvailabilityWaitMs != 0) {
+      // this is the first point in time since the last wait that a chunk became available for filling.
+      waitTimeForNextChunkAvailabilityMs += time.milliseconds() - startTimeForChunkAvailabilityWaitMs;
```

Comment:
yeah, this should be `=` now. `+=` is not incorrect, as this will get updated only once per chunk. This was `+=` due to the earlier implementation where we were calculating this  as an overall value.